diff --git a/python/sglang/srt/managers/scheduler.py b/python/sglang/srt/managers/scheduler.py
--- a/python/sglang/srt/managers/scheduler.py
+++ b/python/sglang/srt/managers/scheduler.py
@@ -705,4 +705,5 @@ class Scheduler(
         self.is_hybrid_ssm = (
             self.tp_worker.model_runner.hybrid_gdn_config is not None
             or self.tp_worker.model_runner.mamba2_config is not None
+            or self.tp_worker.model_runner.rwkv7_config is not None
         )
diff --git a/python/sglang/srt/server_args.py b/python/sglang/srt/server_args.py
--- a/python/sglang/srt/server_args.py
+++ b/python/sglang/srt/server_args.py
@@ -1914,7 +1914,8 @@ class ServerArgs:
         elif model_arch in ["RWKV7ForCausalLM", "Rwkv7ForCausalLM"]:
             self._handle_mamba_radix_cache(
                 model_arch=model_arch,
-                support_mamba_cache=False,
+                support_mamba_cache=True,
+                support_mamba_cache_extra_buffer=False,
             )
         elif model_arch in ["NemotronHForCausalLM"]:
             model_config = self.get_model_config()
